### DISCUSSION TITLE
Open questions

### DISCUSSION THREAD

**Discussion by MatejVukosav** [calimero-network/core#699] [Starknet Hackathon]
Questions about task

**Comment by jaswinder6991** [comment]
I don't see Calimero bounty details on the redacted page - https://redacted.devpost.com/ 

Could you share more on the different tracks, bounties and example solutions or requirements.

**Comment by MatejVukosav** [comment]
Hi, @jaswinder6991 , thanks for reporting. We have fixed the issue and Calimero bounties are now visible on the redacted page.

**Comment by kevinjaypatel** [comment]
Hello, I am currently trying to setup my local node with the Calimero network following the setup guide [https://calimero-network.github.io/getting-started/setup](https://github.com/orgs/calimero-network/discussions/url). I started both the coordinator and peer nodes.

I am having trouble setting up the Admin Dashboard.

- I visit https://calimero-network.github.io/admin-dashboard/
- Set the correct node url
- Continue with Metamask wallet
- Sign the authentication transaction

Upon completing the last step, I am routed to the http://localhost:2428/admin-dashboard/identity
However, the page shows , "Failed to extract authentication headers."

I am using the Ethereum Mainet

**Reply by chefsale** [reply]
cc: @MatejVukosav @xilosada @alenmestrov @frdomovic Can you guys help?

**Reply by alenmestrov** [reply]
hi @kevinjaypatel,

We are unable to reproduce the issue on our end. One possibility could be that we are working from a different commit or a more recent version of the code. Could you please confirm which commit or branch you're using? This will help us better support you.
If you're using an older version, updating to the latest one might resolve the issue, but if it's still happening, we'd be happy to take another look.

**Reply by kevinjaypatel** [reply]
Hi @alenmestrov, 

I believe it might've been this commit (https://github.com/calimero-network/core/pull/721). I am working from the master branch. I pulled the recent commits and will try again. 

I noticed Metamask was behaving strangely yesterday. When I would login to my wallet, the account page would have a refresh symbol persisting for a while. I think this may had implications fetching the headers on the admin dashboard side. 

Nonetheless, I will give the new version of `/core` a go, I was also going to try connecting to a different wallet like `Starknet` perhaps.

**Reply by alenmestrov** [reply]
HI @kevinjaypatel,

thank you for giving it a try!

Please let us know how it goes with the latest version of /core repository.
In regards to the mentioned PR, this was implemented so that the users can have an Admin Dashboard out of the box with binary. It shouldn't affect accessing the node from the Admin Dashboard deployed on GitHub pages. 
If the issue persists please let us know so that we can investigate further.

And thank you for notifying us!

**Reply by kevinjaypatel** [reply]
Hi @alenmestrov, 

I forked the latest version `core` and followed the [setup guide](https://calimero-network.github.io/getting-started/setup/)  

Additionally, I was able to setup my admin dashboard without any trouble seeing my root key. That was the problem I was having yesterday, in place of seeing my root key the dashboard stated, "Failed to extract authentication headers." Now it appears that is not the case

Thank you! I'm looking forward to continue learning Calimero

**Comment by kevinjaypatel** [comment]
Hello, I am trying to create a context for an app using the "Dev Mode" feature in meroctl CLI following the docs here:
[How to Use Dev Mode](https://calimero-network.github.io/build/dev-mode/#how-to-use-dev-mode)

I have both my coordinator and peer node running on separate terminals.
I try running this command: `cargo run -p meroctl -- --home data --node-name node1 context create --watch ./apps/only-peers/res/only_peers.wasm`

The output: 
```
Error:
   0: Config file does not exist

Location:
   crates/meroctl/src/common.rs:67
```

**Reply by petarjuki7** [reply]
Hi @kevinjaypatel,

could you please just verify that there is a `data/` directory where you have initialised your nodes?

Maybe you can try putting the absolute path (specified from the root directory) to the directory where your nodes are after the `--home` flag, and then giving the name of the node (node1 in your case) to the `--node-name` flag like you did already.

**Reply by kevinjaypatel** [reply]
Hi @petarjuki7 , 

Yes, verified there is a `/data` directory in the root of my core repo.

I was able to create the app context, it appears the error was caused because a path to the nodes `config.toml` is expected in place of the `--node-name`. 

Running the following command resolved the issue: 
`cargo run -p meroctl -- --home data --node-name ./data/node1/ context create  --watch ./apps/only-peers/res/only_peers.wasm`

**Comment by kevinjaypatel** [comment]
Hello, 

Currently, I can run a node and send commands to the node using the interactive cli with my terminal, but ideally I want to communicate with the node (daemon) through another interface. 

In particular, I want to communicate with the Node through a CLI wrapper. 
Assuming my client (i.e. CLI wrapper) is on the same network, is this possible? 

Best regards, 
Kevin

**Reply by MatejVukosav** [reply]
We have `merod` and meroctl which can be used outside node terminal. 
`merod` is used to setup the node and `meroctl` is to perform different operations on the node functionalities like `context` and `application`

links: https://calimero-network.github.io/getting-started/setup
https://calimero-network.github.io/build/dev-mode

**Comment by MatejVukosav** [comment]
Hackathon has finished
